editor/components/block-edit/context.js

```diff
@@ -59,3 +60,44 @@ export const ifBlockEditSelected = createHigherOrderComponent( ( OriginalCompone
 		</Consumer>
 	);
 }, 'ifBlockEditSelected' );
+
+/**
+ * A Higher Order Component used to render conditionally the wrapped
+ * component only when the BlockEdit has selected state set or it is
+ * the first block in a multi selection of all one type of block..
+ *
+ * @param {Component} OriginalComponent Component to wrap.
+ *
+ * @return {Component} Component which renders only when the BlockEdit is selected or it is the first block in a multi selection.
+ */
+
+const isFirstOrOnlyBlockSelectedHOC = createHigherOrderComponent( ( OriginalComponent ) => {
+	return ( props ) => {
+		return (
+			<Consumer>
+				{ ( { isSelected, clientId } ) => ( isSelected || ( clientId === props.getFirstMultiSelectedBlockClientId && props.allSelectedBlocksOfSameType ) ) && (
+					<OriginalComponent { ...props } />
+				) }
+			</Consumer>
+		);
+	};
+}, 'isFirstOrOnlyBlockSelected' );
+
+export const isFirstOrOnlyBlockSelected = ( component ) => {
+	return compose( [
+		withSelect( ( select ) => {
+			const {
+				getMultiSelectedBlocks,
+				getFirstMultiSelectedBlockClientId,
+				isMultiSelecting,
+			} = select( 'core/editor' );
+			const allSelectedBlocksOfSameType = uniq( getMultiSelectedBlocks().map( ( { name } ) => name ) ).length === 1;
+			return {
+				getFirstMultiSelectedBlockClientId: getFirstMultiSelectedBlockClientId(),
+				isSelecting: isMultiSelecting(),
+				selectedBlocks: getMultiSelectedBlocks(),
+				allSelectedBlocksOfSameType,
+			};
+		} ),
+	] )( isFirstOrOnlyBlockSelectedHOC( component ) );
```

Comment:
This is what `compose` serves to eliminate, to rewrite:

```js
a( b( c( d( Component ) ) ) );
```

...as:

```js
compose( [ a, b, c, d ] )( Component );
```

In other words, `isFirstOrOnlyBlockSelectedHOC` could be added as an entry of the `compose` argument array.